### PR TITLE
bpo-46443: deepfreeze use preallocated small ints

### PR DESCRIPTION
**Pull request by kumaraditya303**
https://bugs.python.org/issue46443

### CODE REVIEW

**Comment by kumaraditya303** [comment]
It is an implementation detail so skip news.

**Comment by gvanrossum** [comment]
> It is an implementation detail so skip news.

I don't think that's how we usually do this. We skip news for things like fixing typos in comments, but almost anything else gets a new entry. Maybe you're confused with "What's New in 3.11"? That definitely only should get the major visible items.

File: Tools/scripts/deepfreeze.py
Comment (gvanrossum) [inline Tools/scripts/deepfreeze.py:319]: (I wish there was a Python API to find out these boundaries. But there isn't one, so I'm okay with writing it this way.)

**Review by gvanrossum — COMMENTED** [review]
LG except I would like to see a one-line news item.

Did you check how much space this saves? (Probably nut that much since ints are small.)

**Comment by kumaraditya303** [comment]
> > It is an implementation detail so skip news.
> 
> I don't think that's how we usually do this. We skip news for things like fixing typos in comments, but almost anything else gets a new entry. Maybe you're confused with "What's New in 3.11"? That definitely only should get the major visible items.

Ok, I'll add a news fragment. Should I create a bpo for this now ?

**Comment by gvanrossum** [comment]
> Should I create a bpo for this now ?

Yeah.

**Comment by gvanrossum** [comment]
I recommend that you put a little more effort in writing the initial comment for the bpo issues you create. E.g. not just what to do, but why it's worth doing and what impact you expect.

File: Tools/scripts/deepfreeze.py
Comment (kumaraditya303) [reply Tools/scripts/deepfreeze.py:319]: Yes, there isn't any API for this so just had to look up the values in headers.

**Review by gvanrossum — APPROVED** [review]
Thanks! This was indeed super-simple. :-)

**Comment by kumaraditya303** [comment]
I would also do this for empty bytes objects as that also has a global singleton but separately .

**Comment by gvanrossum** [comment]
> I would also do this for empty bytes object as that also has a global singleton but separately.

Good idea. Maybe you can just add that to this PR, it's small enough.

**Comment by kumaraditya303** [comment]
> > It is an implementation detail so skip news.
> 
> I don't think that's how we usually do this. We skip news for things like fixing typos in comments, but almost anything else gets a new entry. Maybe you're confused with "What's New in 3.11"? That definitely only should get the major visible items.

I was confused by this PR as it skipped news https://github.com/python/cpython/pull/30092

**Review by gvanrossum — APPROVED** [review]
Great! Now just waiting for CI to finish...

**Comment by kumaraditya303** [comment]
> Great! Now just waiting for CI to finish...

Automerge :)

**Comment by gvanrossum** [comment]
> I was confused by this PR as it skipped news #30092

Different core devs have different standards...

**Comment by gvanrossum** [comment]
I'm the bot. :-)

**Pull request merged**
